Assets/MRTK/Tests/PlayModeTests/SolverTests.cs

```diff
@@ -352,6 +355,144 @@ public IEnumerator TestHandConstraintPalmUpSolverPlacement()
+        {
+
+            // Instantiate our test GameObject with solver.
+            var testObjects = InstantiateTestSolver<HandConstraintPalmUp>();
+            testObjects.handler.TrackedTargetType = TrackedObjectType.HandJoint;
+            testObjects.handler.TrackedHandness = Handedness.Both;
+
+            var manipHandler = testObjects.target.AddComponent<ManipulationHandler>();
+            manipHandler.HostTransform = testObjects.target.transform;
+            manipHandler.SmoothingActive = false;
+
+            // add near interaction grabbable to be able to grab the cube with the simulated articulated hand
+            testObjects.target.AddComponent<NearInteractionGrabbable>();
+
+
+            var boxCollider = testObjects.target.AddComponent<BoxCollider>();
+            boxCollider.size = new Vector3(10f, 10f, 10f);
+
+            var handConstraintSolver = (HandConstraintPalmUp)testObjects.solver;
+            handConstraintSolver.FollowHandUntilFacingCamera = true;
+            handConstraintSolver.UseGazeActivation = true;
+
+            // Ensure that FacingCameraTrackingThreshold is greater than FollowHandCameraFacingThresholdAngle
+            Assert.AreEqual(handConstraintSolver.FacingCameraTrackingThreshold - handConstraintSolver.FollowHandCameraFacingThresholdAngle > 0, true);
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+
+            TestUtilities.AssertAboutEqual(testObjects.target.transform.position, Vector3.zero, "HandConstraintPalmUp solver did not start at the origin");
+
+            var cameraTransform = CameraCache.Main.transform;
+
+            // Place hand 1 meter in front of user, 50 cm below eye level
+            var handTestPos = cameraTransform.position + cameraTransform.forward + DetermineHandOriginPositionOffset(HandConstraint.SolverSafeZone.UlnarSide, Handedness.Left);
+
+            var cameraLookVector = (handTestPos - cameraTransform.position).normalized;
+
+            // Generate hand rotation with hand palm facing camera
+            var handRotation = Quaternion.LookRotation(cameraTransform.up, cameraLookVector);
+
+            // Add a right hand.
+            var leftHand = new TestHand(Handedness.Left);
+            yield return leftHand.Show(handTestPos);
+            yield return leftHand.SetRotation(handRotation);
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+
+            TestHand rightHand = new TestHand(Handedness.Right);
+            yield return rightHand.Show(new Vector3(0, 0, 0.5f));
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.OpenSteadyGrabPoint);
+            yield return rightHand.MoveTo(testObjects.target.transform.position, 5);
+            
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+            
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            testObjects.handler.UpdateSolvers = false;
```

Comment:
Earnestly it was because I wasn't sure how to properly implement an event listener to that event in test code; though I remember being a bit hurried when implementing this test initially. I'll look into this.